crypt/key.go

```diff
@@ -123,6 +141,31 @@ func (k *Key) Decrypt(s string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("invalid decryption payload: %v", err)
 	}
+	if len(buf) < 1 {
+		return "", ErrPayLoadTooShort
+	}
+	// Some implementations use a version-prefixed cipher text.
+	switch buf[0] {
+	case byte(1):
+		bytes, err := k.decryptSecretbox(buf[1:])
+		if err != nil {
+			// In order to handle the (unlikely but possible) case
+			// where a versionless payload *just happens* to start
+			// with byte(1), we must also try the fallback here.
+			goto fallback
+		}
+		return string(bytes[:]), nil
```

Comment:
Instead of using a `goto`, you could simply invert the conditional and return here `if err == nil`.

Reply:
Ah, good catch. Fixed.